packages/mobile/src/services/track-download.ts

```diff
@@ -74,59 +78,139 @@ const download = async ({
-        // with the share sheet, then delete the cached copy of the track.
-        fileCache: true,
-        path: filePath
-      }),
-      onFetchComplete: async (fetchRes) => {
-        await Share.share({
-          url: fetchRes.path()
-        })
-        fetchRes.flush()
-      }
+  const audiusDirectory =
+    RNFetchBlob.fs.dirs.DocumentDir + '/' + audiusDownloadsDirectory
+  const onFetchComplete = async (path: string) => {
+    dispatch(downloadFinished())
+    await Share.share({
+      url: path
     })
+  }
+
+  if (Platform.OS === 'ios') {
+    if (files.length === 1) {
+      const { url, filename } = files[0]
+      downloadOne({
+        fileUrl: url,
+        filename,
+        directory: audiusDirectory,
+        getFetchConfig: (filePath) => ({
+          // On iOS fetch & cache the track, let user choose where to download it
+          // with the share sheet, then delete the cached copy of the track.
+          fileCache: true,
+          path: filePath
+        }),
+        onFetchComplete
+      })
+    } else {
+      downloadMany({
+        files,
+        directory: audiusDirectory + '/' + rootDirectoryName,
+        getFetchConfig: (filePath) => ({
+          // On iOS fetch & cache the track, let user choose where to download it
+          // with the share sheet, then delete the cached copy of the track.
+          fileCache: true,
+          path: filePath
+        }),
+        onFetchComplete
+      })
+    }
   } else {
-    download({
-      fileUrl,
-      fileName,
-      directory: RNFetchBlob.fs.dirs.DownloadDir,
-      getFetchConfig: (filePath) => ({
-        // On android save to FS and trigger notification that it is saved
-        addAndroidDownloads: {
-          description: trackName,
-          mediaScannable: true,
-          mime: 'audio/mpeg',
-          notification: true,
-          path: filePath,
-          title: trackName,
-          useDownloadManager: true
-        }
+    if (files.length === 1) {
+      const { url, filename } = files[0]
+      downloadOne({
+        fileUrl: url,
+        filename,
+        directory: RNFetchBlob.fs.dirs.DownloadDir,
+        getFetchConfig: (filePath) => ({
+          // On android save to FS and trigger notification that it is saved
+          addAndroidDownloads: {
+            description: filename,
+            mediaScannable: true,
+            mime: 'audio/mpeg',
+            notification: true,
+            path: filePath,
+            title: filename,
+            useDownloadManager: true
+          }
+        })
       })
-    })
+    } else {
+      downloadMany({
+        files,
+        directory: RNFetchBlob.fs.dirs.DownloadDir,
+        getFetchConfig: (filePath) => ({
+          // On android save to FS and trigger notification that it is saved
+          addAndroidDownloads: {
+            description: rootDirectoryName,
+            mediaScannable: true,
+            mime: 'audio/mpeg',
```

Comment:
same question as above about `audio/mpeg` mime type

Reply:
good callout, not sure but it appears to be downloading the lossless files correctly for me.

Reply:
Created [a linear](https://linear.app/audius/issue/PAY-2414/investigate-audiompeg-mime-type-in-mobile-downloads) to come back to this.